dashboard/app/views/levels/_level_group.haml

```diff
@@ -25,6 +25,14 @@
   - parsed = JSON.parse(@last_attempt || '{}')
   - page.levels.each_with_index do |level, index|
 
+    -# Show any text from an external level.
+
+    - data["texts"].select {|text| text["index"] == page.offset + index}.each do |text_level_info|
```

Comment:
Looks like this may throw when `data["texts"]` is nil on `/s/allthethings/stage/22/puzzle/2/page/2`.  If you don't repro I may have a problem with how I seeded scripts.